### PR TITLE
feat: Support sending logs via OTLP

### PR DESCRIPTION
**Pull request by popzxc**
## What ❔

Makes it possible to export logs via OTLP.
This doesn't disable exporting logs to stdout, which is extra valuable when working with localhost observability stack.
Also, it improves handling of `ServiceDescriptor` so that it has some somewhat sane values by default (setting data locally wasn't convenient so far).

## Why ❔

Being able to collect logs to any OTLP collector.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

**Review by itegulov — APPROVED** [review]
❤️

File: core/lib/vlog/src/opentelemetry/mod.rs [hunk 1]
```diff
@@ -14,67 +14,88 @@ use tracing_subscriber::{registry::LookupSpan, EnvFilter, Layer};
...
+            k8s_pod_name: env_or(Self::K8S_POD_NAME_ENV_VAR, Self::DEFAULT_K8S_POD_NAME),
+            k8s_namespace_name: env_or(
+                Self::K8S_NAMESPACE_NAME_ENV_VAR,
+                Self::DEFAULT_K8S_NAMESPACE_NAME,
+            ),
+            service_name: env_or(Self::SERVICE_NAME_ENV_VAR, Self::DEFAULT_SERVICE_NAME),
+        }
     }
 
     pub fn with_k8s_pod_name(mut self, k8s_pod_name: Option<String>) -> Self {
```
[the remark this answers is not in the document]
Comment (popzxc) [reply core/lib/vlog/src/opentelemetry/mod.rs:72]: It's a somewhat counter-intuitive, but it's actually pretty convenient for applying configs. In some external config, you often _maybe_ have an override, e.g. `field_override: Option<String>`. If you would accept `String` here, then calling this method would look like `if let Some(override) = config.field_override { descriptor = descriptor.with_k8s_pod_name(override) }` and so on. With `Option`, you can pass whatever you have. And if your override is not an `Option`, you can just wrap it in `Some`.

tl;dr: easier for the caller to use it.

**Comment by popzxc** [comment]
@EmilLuta thanks for the comments; I will resolve them in a follow-up -- I really need to merge this PR ASAP.
